src/couch/src/couch_httpd_auth.erl

```diff
@@ -186,6 +188,25 @@ proxy_auth_user(Req) ->
             end
     end.
 
+jwt_authentication_handler(Req) ->
+    case {config:get("jwt_auth", "secret"), header_value(Req, "Authorization")} of
+        {undefined, _} -> Req;
+        {_, undefined} -> Req;
+        {Secret, "Bearer " ++ Jwt} ->
+            RequiredClaims = re:split(config:get("jwt_auth", "required_claims", "sub, exp"), "\s*,\s*"),
+            AllowedAlgorithms = re:split(config:get("jwt_auth", "allowed_algorithms", "HS256"), "\s*,\s*"),
+            TimestampLeniency = config:get_integer("jwt_auth", "timestamp_leniency", "0"),
+            case jwtf:decode(?l2b(Jwt), RequiredClaims, fun(_,_) -> Secret end) of
+                {ok, {Claims}} ->
+                    {_, User} = lists:keyfind(<<"sub">>, 1, Claims),
```

Comment:
since `sub` is mandatory, an admin that doesn't include it in the `required_claims` property is just going to cause a `badarg` here.

Reply:
Are you suggesting we prepend/append `<<"sub">>` to the config?

Reply:
that seems the right move here. noting also that we do nothing with AllowedAlgorithms or TimestampLeniency as yet.

Reply:
(noting also that we'll just pull the jwtf code directly into this project.)

Reply:
Does erlang or couchdb have a fancy utility method to intersect two lists?

Reply:
Done!